.github/workflows/check_build.yml

```diff
@@ -76,6 +76,8 @@ jobs:
         with:
           path: '~/.konan/**'
           key: konan-ios
+      - name: Xcode cache
+        uses: mikehardy/buildcache-action@v2
```

Comment:
@dfernandezm I played around with this as you can see 😂 

There were several issues:
- for some reason `mikehardy/buildcache-action@v1` stopped working on our `macos-11` environment. The build was complaining that it was compiled for macos12 🤷 
- so I updated to `macos-12` which required a few other updates too
- Xcode 13.0 no longer worked so I had to update it to 13.3
- this also made the iOS simulator fail, so it had to be updated to 15.4

Looks like all is green now with the cache, so I think we can merge this when ready! Let me know if you have questions 👍 

This was an annoying one 😅

Reply:
Awesome, I'll merge